FWCore/Modules/test/testGenericConsumer.py

```diff
@@ -0,0 +1,44 @@
+import FWCore.ParameterSet.Config as cms
+
+process = cms.Process("TEST")
+
+process.load("FWCore.Framework.test.cmsExceptionsFatal_cff")
+process.load("FWCore.MessageService.MessageLogger_cfi")
+process.MessageLogger.cerr.INFO.limit = 10000000
+
+process.maxEvents = cms.untracked.PSet(
+    input = cms.untracked.int32(5)
+)
+
+process.source = cms.Source("EmptySource")
+
+process.thing = cms.EDProducer("ThingProducer")
+
+process.notRunningThing = cms.EDProducer("ThingProducer")
+
+process.anotherThing = cms.EDProducer("ThingProducer")
+
+process.otherThing = cms.EDProducer("OtherThingProducer",
+    thingTag = cms.InputTag('anotherThing'),
+    transient = cms.untracked.bool(True)
+)
+
+# a ConditionalTask makes the modules that are consumed on a path to be scheudled (only) in that path
+process.task = cms.ConditionalTask(
```

Comment:
I'd like this to be `Task`. In practice `ConditionalTask` does achieve the job of checking that the `consumes()` calls of the `GenericConsumer` get registered correctly, so I consider this to be good enough (although I might change it later).

Reply:
Ah, yes, `CondutionalTask` is just a leftover from the previous attempt.